tests/test/slg.rs

```diff
@@ -294,7 +294,8 @@ fn only_draw_so_many_blow_up() {
 
         goal {
             exists<T> { T: Foo }
-        } fixed 2 with max 10 {
+        } fixed 767 with max 10 {
```

Comment:
what's up with this? =)

Reply:
I honestly am not sure. Was hoping maybe you had some insight?

Reply:
OK :) I'll take a look

Reply:
OK -- I THINk I figured out the problem here. It's sort of subtle. The problem has to do with this rather ill-documented mechanism of `any_future_answer`. The problem, I believe, is that the `any_future_answer` method is being invoked with substitutions that include inference variables but some of those inference variables are not fully normalized. This leads to `may_invalidate` giving the wrong results -- and what's worse, it doesn't have access to the inference tables it needs.

The easiest fix would be to canonicalize before invoking the callback in `any_future_answer`, but that seems ungreat, since this routine is invoked frequently, and canonicalization is potentially expensive. I have a potential fix that adds a `normalize_subst` routine that we can invoke, but that's actually a similar-ish cost and also a bit of a "footgun", in that we don't want this operation 99% of the time. We could also thread down the inference table, but *that* is a bit annoying since sometimes the substitutions are coming from things that have been canonicalized (which the current code, with its assertions, doesn't seem to be accounting for).

Anyway, my "not great" fix is also not actually seeming to work, so let me make sure I am right before discussing deeper.

Reply:
Eek. I definitely didn't expect that. I'll have to take a look